### PR TITLE
fix(angular/chips): set dirty state and correctly delete entries in arrays

### PR DESCRIPTION
**Pull request by jeripeierSBB**

### CODE REVIEW

File: src/angular/chips/chip.ts [hunk 1]
```diff
@@ -233,17 +233,34 @@ export class SbbChip
...
+
+    if (isArray) {
       if (!currentCollection.includes(this.value)) {
         return;
       }
-      control.patchValue(currentCollection.filter((val) => val !== this.value));
-    } else if (currentCollection instanceof Set) {
+      const indexInChipList = this._chipList?.chips.toArray().indexOf(this);
+      if (
+        typeof indexInChipList !== 'undefined' &&
```
Comment (jeripeierSBB) [inline src/angular/chips/chip.ts:247]: check if it can be shortened

File: src/angular/chips/chip.ts [hunk 2]
```diff
@@ -233,17 +233,34 @@ export class SbbChip
...
+        indexInChipList !== -1 &&
+        currentCollection[indexInChipList] === this.value
+      ) {
+        control.patchValue(currentCollection.filter((val, index) => indexInChipList !== index));
+      } else {
+        control.patchValue(currentCollection.filter((val) => val !== this.value)); // Can delete more than one entry
```
Comment (jeripeierSBB) [inline src/angular/chips/chip.ts:253]: move comment up

**Review by jeripeierSBB — COMMENTED** [review]
TODO: autocomplete example does not reset list entries

**Review by kyubisation — APPROVED** [review]
LGTM 👍

**Review by mhaertwig — APPROVED** [review]
LGTM 👍

**Pull request merged**